core/src/main/java/org/jdbi/v3/core/OnDemandExtensions.java

```diff
@@ -51,22 +52,30 @@
                 return extensionType + "@" + Integer.toHexString(System.identityHashCode(proxy));
             }
 
-            try {
-                if (threadExtension.get() != null) {
-                    return method.invoke(threadExtension.get(), args);
-                }
-                return db.withExtension(extensionType, extension ->
-                        JdbiThreadLocals.invokeInContext(threadExtension, extension,
-                                () -> method.invoke(extension, args)));
-            }
-            catch (InvocationTargetException e) {
-                throw e.getTargetException();
+            if (threadExtension.get() != null) {
+                return invoke(method, threadExtension.get(), args);
             }
+            return db.withExtension(extensionType, extension ->
+                    JdbiThreadLocals.invokeInContext(threadExtension, extension,
+                            () -> invoke(method, extension, args)));
         };
 
         return extensionType.cast(
                 Proxy.newProxyInstance(
                         extensionType.getClassLoader(),
                         new Class[]{extensionType}, handler));
     }
+
+    private static Object invoke(Method method, Object target, Object[] args) {
```

Comment:
\<pedant\>Should this be `target, method, args` instead, to match parameter order elsewhere in the project?\<\/pedant\>